File: client_credentials.php
<?php
define('CLIENT_ID', '');
define('CLIENT_SECRET', '');
define('TOKEN_URL', 'https://allegro.pl/auth/oauth/token');

function getCurl($headers, $content) {
	$ch = curl_init();
	curl_setopt_array($ch, array(
		CURLOPT_URL => TOKEN_URL,
		CURLOPT_HTTPHEADER => $headers,
		CURLOPT_SSL_VERIFYPEER => false,
		CURLOPT_RETURNTRANSFER => true,
		CURLOPT_POST => true,
		CURLOPT_POSTFIELDS => $content
	));
	return $ch;
}

function getAccessToken() 
{
	$authorization = base64_encode(CLIENT_ID.':'.CLIENT_SECRET);
	$headers = array("Authorization: Basic {$authorization}","Content-Type: application/x-www-form-urlencoded");
	$content = "grant_type=client_credentials";
	$ch = getCurl($headers, $content);
	$tokenResult = curl_exec($ch);
    $resultCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
	curl_close($ch);
    if ($tokenResult === false || $resultCode !== 200) {
        exit ("Something went wrong");
    }
	return json_decode($tokenResult)->access_token;
}

function main()
{
    echo "access_token = ", getAccessToken();
}

main();
?>

File: client_credentials.py
import requests
import json

CLIENT_ID = '42c06fb738594d198e571b47d39bdd75'
CLIENT_SECRET = 'ifPKW8LZUNfEKEIhvDaOcMzP4daIhQEcmE7qB9iQMRlxdQKLxBhncLIrUp2piGyC'
TOKEN_URL = "https://allegro.pl/auth/oauth/token"


def get_access_token():
    try:
        data = {'grant_type': 'client_credentials'}
        access_token_response = requests.post(TOKEN_URL, data=data, verify=False, allow_redirects=False, auth=(CLIENT_ID, CLIENT_SECRET))
        return access_token_response
    except requests.exceptions.HTTPError as err:
        raise SystemExit(err)


def main():
    access_token_response = get_access_token()
    tokens = json.loads(access_token_response.text)
    access_token = tokens['access_token']
    print("access token: " + access_token)


if __name__ == "__main__":
    main()

